plugins/auto-sizes/hooks.php

```diff
@@ -139,79 +139,145 @@ function auto_sizes_get_width( string $layout_width, int $image_width ): string
 }
 
 /**
- * Filter the sizes attribute for images to improve the default calculation.
+ * Primes attachment into the cache with a single database query.
  *
- * @since 1.1.0
+ * @since n.e.x.t
  *
- * @param string                                                   $content      The block content about to be rendered.
- * @param array{ attrs?: array{ align?: string, width?: string } } $parsed_block The parsed block.
- * @return string The updated block content.
+ * @param string|mixed $content The HTML content.
+ * @param string       $context Optional. Additional context to pass to the filters.
+ *                              Defaults to `current_filter()` when not set.
+ * @return string The HTML content.
  */
-function auto_sizes_filter_image_tag( string $content, array $parsed_block ): string {
+function auto_sizes_prime_attachment_caches( $content, string $context = null ): string {
+	if ( ! is_string( $content ) ) {
+		return '';
+	}
+	if ( null === $context ) {
+		$context = current_filter();
+	}
```

Comment:
Perhaps my above comments should be adjusted to delete the `$context` param altogether because actually the variable isn't being used at all. So should it just be removed?

```suggestion
	if ( null === $context ) {
		$context = current_filter();
	}
```

Reply:
Also, as it stands right now, the 2nd parameter is never passed in anyway:

```php
add_filter( 'the_content', 'auto_sizes_prime_attachment_caches', 6 );
```